### DISCUSSION TITLE
List supported file systems for csi.storage.k8s.io/fstype

### DISCUSSION THREAD

**Discussion by LukeMauldin** [ceph/ceph-csi#3243] [Q&A]
# Describe the bug #

The documentation states that `csi.storage.k8s.io/fstype` defaults to the file system type `ext4`.  What are the other supported file system values?  Specifically, is `btrfs` a supported value?

# Environment details #

- Image/version of Ceph CSI driver : v3.6.0
- Helm chart version :
- Kernel version : 5.3.18

**Comment by Madhu-1** [comment]
for now ext4 and xfs, buts ext4 is recommended as there is an issue with xfs https://github.com/rook/rook/issues/3132#issuecomment-580508760

**Comment by LukeMauldin** [comment]
Will other file systems such as `btrfs` work or will ceph-csi fail trying to provision them?

**Comment by Madhu-1** [comment]
> Will other file systems such as `btrfs` work or will ceph-csi fail trying to provision them?

We have tested ext4 and xfs, if rbd/kubernetes supports other filesystem types it should be supported with cephcsi also. Please try it out and let us know how it works.

**Comment by LukeMauldin** [comment]
I verified the static provisioned volume with ext4 and that worked.  When I tried with btrfs file system I received the error below.
It seems that only xfs, ext3, and ext4 are supported by this plugin.  Do I need to put in a feature request to support btrfs as well?

 `Warning  FailedMount             1s (x5 over 10s)  kubelet                  MountVolume.MountDevice failed for volume "ceph-rbd-static-pv" : rpc error: code = Internal desc = rpc error: code = Internal desc = need resize check failed on devicePath /dev/rbd1 and staingPath /var/lib/kubelet/plugins/[kubernetes.io/csi/pv/ceph-rbd-static-pv/globalmount/csi-test](http://kubernetes.io/csi/pv/ceph-rbd-static-pv/globalmount/csi-test), error: Could not parse fs info on given filesystem format: btrfs. Supported fs types are: xfs, ext3, ext4

**Comment by LukeMauldin** [comment]
I did some more research and it looks like btrfs support was recently added into Kubernetes mount utilities: https://github.com/kubernetes/mount-utils/commit/f60c4b2fde525ca3ee0712747f041880e508b5fd

I am using an older version of Kubernetes (1.22.x) and so I don't think btrfs is supported in my version.

**Reply by Madhu-1** [reply]
@LukeMauldin it is only a client package change, give a try with `quay.io/cephcsi/cephcsi:canary` image which uses kubernetes 1.24

**Comment by LukeMauldin** [comment]
If I upgrade to the canary version and I have problems, will it be safe to downgrade back to a 3.6.x release? Any idea when a version of cephcsi that uses Kuberentes 1.24 will be released to stable?

**Reply by Madhu-1** [reply]
canary is not something to use for production clusters. its only for testing clusters. (canary image is built whenever a PR is merged in master /devel branch). If you want a stable release need to wait for 3.7.0 release (may be in 1 or 2 months)

**Comment by LukeMauldin** [comment]
Also for the static rbd images, where would I specific mount options such as noatime?

**Reply by Madhu-1** [reply]
mount options are need to be added to the storageclass and kubernetes will take care of sending mount options in NodeStageRequest for cephcsi
